src/bmp/decoder.rs

```diff
@@ -670,42 +700,46 @@ impl<R: Read + Seek> BmpDecoder<R> {
                 32 if self.add_alpha_channel => ImageType::RGBA32,
                 32 => ImageType::RGB32,
                 _ => {
-                    return Err(ImageError::FormatError(format!(
-                        "Invalid RGB bit count {}",
-                        self.bit_count
+                    return Err(ImageError::Decoding(DecodingError::with_message(
+                        ImageFormat::Bmp.into(),
+                        format!("Invalid RGB bit count {}", self.bit_count),
                     )))
                 }
             },
             1 => match self.bit_count {
                 8 => ImageType::RLE8,
                 _ => {
-                    return Err(ImageError::FormatError(
-                        "Invalid RLE8 bit count".to_string(),
-                    ))
+                    return Err(ImageError::Decoding(DecodingError::with_message(
+                        ImageFormat::Bmp.into(),
+                        "Invalid RLE8 bit count".to_owned(),
+                    )))
                 }
             },
             2 => match self.bit_count {
                 4 => ImageType::RLE4,
                 _ => {
-                    return Err(ImageError::FormatError(
+                    return Err(ImageError::Decoding(DecodingError::with_message(
+                        ImageFormat::Bmp.into(),
                         "Invalid RLE4 bit count".to_string(),
-                    ))
+                    )))
                 }
             },
             3 => match self.bit_count {
                 16 => ImageType::Bitfields16,
                 32 => ImageType::Bitfields32,
                 _ => {
-                    return Err(ImageError::FormatError(
+                    return Err(ImageError::Decoding(DecodingError::with_message(
+                        ImageFormat::Bmp.into(),
                         "Invalid bitfields bit count".to_string(),
-                    ))
+                    )))
                 }
             },
             // PNG and JPEG not implemented yet.
             _ => {
-                return Err(ImageError::UnsupportedError(
+                return Err(ImageError::Decoding(DecodingError::with_message(
+                    ImageFormat::Bmp.into(),
                     "Unsupported image type".to_string(),
```

Comment:
Which should be used, `DecodingError` or `UnsupportedError`?

Possible candidate:

```rust
                return Err(ImageError::Unsupported(
                    UnsupportedError::from_format_and_kind(
                        ImageFormat::Bmp.into(),
                        UnsupportedErrorKind::GenericFeature(format!(
                            "image type {}",
                            image_type_u32
                        )),
                    ),
                ));
```

Reply:
This should be split into two: `UnsupportedError` for values `BI_JPEG = 4` and `BI_PNG = 5` and then a `DecodingError` for all other values.